app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,433 @@
+            onChange={(e: SyntheticInputEvent<>) =>
+              this.setState({ showOnlySharedExperiences: e.target.checked })
+            }
+          >
+            Show Only Shared Experience Domains
+          </Checkbox>
+        ) : null}
+        <Table
+          size="small"
+          dataSource={mutlipleUsers ? multipleUsersEntries : singleUsersEntries}
+          rowKey="domain"
+          pagination={false}
+          scroll={scroll}
+          className="user-experience-table"
+        >
+          <Column
+            title="Experience Domain"
+            key="domain"
+            dataIndex="domain"
+            width={mutlipleUsers ? "30%" : "40%"}
+          />
+          {mutlipleUsers ? (
+            <Column
+              title="Current Experience Value"
+              width="30%"
+              render={(record: MultipleUserTableEntry) =>
+                // eslint-disable-next-line no-nested-ternary
+                record.highestValue === -1 && record.lowestValue === -1
+                  ? ""
+                  : record.highestValue === record.lowestValue
+                    ? record.highestValue
+                    : `varying from ${record.lowestValue} to ${record.highestValue}`
+              }
+            />
+          ) : null}
+          <Column
+            title="Experience Value"
+            key="value"
+            width={mutlipleUsers ? "25%" : "40%"}
+            render={record => {
+              const index = mutlipleUsers
+                ? this.state.multipleUsersEntries.findIndex(entry => entry.domain === record.domain)
+                : this.state.singleUsersEntries.findIndex(entry => entry.domain === record.domain);
+              return (
+                <span>
+                  <InputNumber
+                    value={
+                      mutlipleUsers
+                        ? this.state.multipleUsersEntries[index].value
+                        : this.state.singleUsersEntries[index].value
+                    }
+                    onChange={value => this.setValueOfEntry(index, value)}
+                  />
+                  {(mutlipleUsers && record.value !== record.lowestValue) ||
+                  this.recordModifiedAndExistedBefore(record) ? (
+                    <Tooltip placement="top" title="Revert Changes">
+                      <Icon
+                        style={{ marginLeft: 15 }}
+                        className="clickable-icon"
+                        type="rollback"
+                        onClick={() => this.revertChangesOfEntry(index)}
+                      />
+                    </Tooltip>
+                  ) : null}
+                </span>
+              );
+            }}
+          />
+          <Column
+            title="Delete Entry"
+            key="removed"
+            width={mutlipleUsers ? "15%" : "20%"}
+            render={record => {
+              const index = singleUsersEntries.findIndex(entry => entry.domain === record.domain);
```

Comment:
Doesn't this break if `singleUsersEntries` is empty (which is the case when we are editing multiple users)?

Also: you are looking up the index which corresponds to `record.domain` and in `removeEntryFromTable` you are looking up `domain` for `index`. So, the look up goes like this: `domain -> index -> domain`. Why don't you pass the domain directly to `removeEntryFromTable`? This would also work for both single and multiple editing mode.